Dutchskull.Aspire.AppHost.Unity3D/Dutchskull.Aspire.AppHost.Unity3D/UnityAspireExtensions.cs

```diff
@@ -56,122 +60,265 @@ public static IResourceBuilder<UnityProjectResource> AddUnityProject(
+            HealthCheckService healthCheck = services.GetRequiredService<HealthCheckService>();
 
-                HealthCheckService healthCheck = services.GetRequiredService<HealthCheckService>();
+            while (!cancellationToken.IsCancellationRequested && (DateTime.UtcNow - start) < overallTimeout)
+            {
+                HealthReport health =
+                    await healthCheck.CheckHealthAsync(cancellationToken).ConfigureAwait(false);
 
-                while (!cancellationToken.IsCancellationRequested && (DateTime.UtcNow - start) < overallTimeout)
+                if (health.Entries[healthCheckEditorKey].Status == HealthStatus.Healthy)
                 {
-                    HealthReport health = await healthCheck.CheckHealthAsync(cancellationToken).ConfigureAwait(false);
+                    break;
+                }
 
-                    if (health.Status == HealthStatus.Healthy)
-                    {
-                        break;
-                    }
+                await Task.Delay(pollInterval, cancellationToken).ConfigureAwait(false);
+            }
 
-                    await Task.Delay(pollInterval, cancellationToken).ConfigureAwait(false);
-                }
+            try
+            {
+                bool runSucceeded = await controlClient
+                    .StartProjectAsync(unityResource.ControlUrl, sceneName, cancellationToken)
+                    .ConfigureAwait(false);
 
-                try
+                if (runSucceeded)
                 {
-                    bool runSucceeded = await controlClient.StartProjectAsync(unityResource.ControlUrl, sceneName, cancellationToken).ConfigureAwait(false);
-
-                    if (runSucceeded)
-                    {
-                        await notifications.PublishUpdateAsync(resource, s => s with
-                        {
-                            StartTimeStamp = DateTime.UtcNow,
-                            State = KnownResourceStates.Running,
-                            Urls = s.Urls.Add(new UrlSnapshot(unityResource.ControlUrl.ToString(), unityResource.ControlUrl.ToString(), false))
-                        }).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        await notifications.PublishUpdateAsync(resource, s => s with
-                        {
-                            StartTimeStamp = DateTime.UtcNow,
-                            State = "StartedButRunFailed",
-                        }).ConfigureAwait(false);
-                    }
+                    await notifications
+                        .PublishUpdateAsync(
+                            resource,
+                            s => s with
+                            {
+                                StartTimeStamp = DateTime.UtcNow,
+                                State = KnownResourceStates.Running,
+                                Urls = s.Urls.Add(
```

Comment:
I found this out while doing the review. This is going to a seperate issue. You can ignore this.

<img width="360" height="221" alt="Image" src="https://github.com/user-attachments/assets/a1d02a5e-d50b-4322-8f2a-4dbcf3fe79ca" />